### PR TITLE
Feat/get supported regions

### PR DESCRIPTION
**Pull request by imperialis**
### What does this PR do?

This PR implements an endpoint for getting the list of supported regions:

- **Added GET /api/v1/regions Endpoint:** Returns the list of supported regions.
- **Response Handling:** Provides appropriate responses for missing or invalid refresh tokens.
- **Error Handling:** Returns specific HTTP status codes and messages based on authorization status.

### How should this be manually tested?

1. **Register a User:**
   - Use the `POST /api/v1/auth/register` endpoint to register a new user.

2. **Login User:**
   - Make a `POST` request to `/api/v1/auth/login` with the registered user's password and email in the request body.
   - Check the response for appropriate status codes and messages.

3. **Get Supported Regions:**
   - Make a `GET` request to `/api/v1/regions` with a valid token.
   - Verify the response contains the list of supported regions.

4. **Edge Cases:**
   - **Invalid Token:** Test with an invalid token to ensure the endpoint returns a `401 Unauthorized` status with the correct message.
   - **Missing Token:** Test without a token to ensure the endpoint returns a `401 Unauthorized` status with the correct message.

### Checklist

- [x] Implemented `GET /api/v1/regions` endpoint in the `regions.controller.ts`.
- [x] Added `findAll` method in the `RegionsService`.
- [x] Modified `regions.controller.ts` to provide appropriate responses for various cases.
- [x] Tested endpoint functionality to ensure it correctly handles valid and invalid tokens.
- [x] Added appropriate HTTP status codes and response messages for edge cases.
- [x] Added tests for this endpoint.

### Response Examples

**Success:**
```json
{
  "status": "success",
  "data": {
    "regions": [
      {
        "id": "string",
        "region": "string"
      }
    ]
  }
}
```

**Error:**
```json
{
  "status": "error",
  "message": "You are not authorized for this action",
  "status_code": 401
}
```

### Link to Issue

Link to this issue: #34

### CODE REVIEW

**Review by davidolanrewaju — COMMENTED** [review]
Please document - [NestJS OpenAPI Documentation](https://docs.nestjs.com/openapi/types-and-parameters).
Also make use of the global entity provided in the project
Screenshot (149)

**Review by davidolanrewaju — APPROVED** [review]

File: src/modules/regions/entities/region.entity.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { Entity, PrimaryGeneratedColumn, Column, CreateDateColumn, UpdateDateColumn } from 'typeorm';
```
Comment (Cobidex) [inline src/modules/regions/entities/region.entity.ts:1]: Please keep your code DRY by extending the base entity class.
Then remove the created_at, updated_at and I'd fields from this definition afterwards

File: .gitignore [hunk 2]
```diff
@@ -391,3 +391,10 @@ dist
 /.env
 /.env.*.local
 *._local.ts
+
+# Ignore specific files
+/src/database/data-source.ts
+
+# Package files
+/package-lock.json
+/package.json
```
Comment (TheCodeGhinux) [inline .gitignore:400]: Why are we adding these

File: src/modules/regions/entities/region.entity.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+@Entity()
+export class Regions {
+  @PrimaryGeneratedColumn('uuid')
+  id: string;
+
```
Comment (TheCodeGhinux) [inline src/modules/regions/entities/region.entity.ts:7]: Use AbstractBaseEntity

**Comment by TheCodeGhinux** [comment]
Fix conflicts
